formats/polus-feather-to-tabular-plugin/src/main.py

```diff
@@ -0,0 +1,137 @@
+from pathlib import Path
+import os
+import argparse
+import logging
+from concurrent.futures import ProcessPoolExecutor
+from multiprocessing import cpu_count
+import filepattern
+import pyarrow as pa
+import pyarrow.feather as pf
+import pyarrow.parquet as pq
+import pyarrow.csv as csv
+import shutil
+
+
+# Import environment variables
+POLUS_LOG = getattr(logging,os.environ.get('POLUS_LOG','INFO'))
+FILE_EXT = os.environ.get('POLUS_TAB_EXT')
+# FILE_EXT = FILE_EXT if FILE_EXT is not None else '.csv'
+
+#Set number of processors for scalability
+NUM_CPUS = max(1, cpu_count() // 2)
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(POLUS_LOG)
+
+def remove_files(outDir):
+    logger.info('Removing intermediate files... ')
+    outP = Path(outDir)
+    for file in outP.iterdir():
+        ext = Path(file).suffix
+        if ext == '.feather':
+            os.remove(file)
+    logger.info('Done') 
+    
+def feather_to_tabular(file: Path, filePattern: str, outDir: Path):
+    """Converts feather file into tabular file using pyarrow
+            
+            Args:
+                file (Path): Path to input file.
+                filePattern (str): Filepattern of desired tabular output file.
+                outDir (Path): Path to output directory.              
+            Returns:
+                Tabular File
+                
+    """
+    # Copy file into output directory for WIPP Processing
+    filepath = file.get('file')
+    file_name = Path(filepath).stem
+    logger.info('Feather CONVERSION: Copy %s into outDir for processing...', file_name)
+    output = file_name + ".feather"
+    outputfile = os.path.join(outDir, output)
+    shutil.copyfile(filepath, outputfile)
+    
+    #Get filename for tabular output file
+    pq_file = file_name + ".parquet"
+    csv_file = file_name + ".csv"
+
+    logger.info('Feather CONVERSION: Converting file into PyArrow Table')
+    
+    table = pf.read_table(outputfile)
+
+    if filePattern == ".*.csv":
+        # Streaming contents of Arrow Table into csv
+        logger.info('Feather CONVERSION: converting PyArrow Table into .csv file')
+        os.chdir(outDir)
+        return csv.write_csv(table, csv_file)
+    elif filePattern == ".*.parquet":
+        logger.info('Feather CONVERSION: converting PyArrow Table into .parquet file')
+        os.chdir(outDir)
+        return pq.write_table(table, pq_file)
+    # If neither, log error
+    else:
+        logger.error('Feather CONVERSION Error: This filePattern is not supported in this plugin')
```

Comment:
This is going to be fragile, and won't let us subset data if the user chooses to do so.

Also, it's no ideal to change the current directory from inside the code. The `outDir` should be a part of the output path and file name.